drivers/serial/uart_ns16550.c

```diff
@@ -332,38 +332,12 @@ static int uart_ns16550_configure(const struct device *dev,
 
 	k_spinlock_key_t key = k_spin_lock(&dev_data->lock);
 
-	ARG_UNUSED(dev_data);
-	ARG_UNUSED(dev_cfg);
```

Comment:
I think the change should not affect `ARG_UNUSED()`

Reply:
> I think the change should not affect `ARG_UNUSED()`

dev_data and dev_cfg are used without any conditions so I thought ARG_UNUSED() is not needed.